tests/contrib/integration_registry/test_contrib_versions.py

```diff
@@ -37,67 +91,123 @@ def get_registry_min_version(registry_entry: dict) -> Optional[Version]:
-
-    def test_documented_integrations_are_tested(self, documented_versions: Dict[str, str], registry_data: list[dict]):
-        """Test that documented external integrations are tested."""
-        registry_by_name = {entry["integration_name"]: entry for entry in registry_data}
-
-        untested = [
-            name
-            for name in documented_versions
-            if (
-                name in registry_by_name
-                and not registry_by_name[name].get("is_tested", False)
-                and name not in EXCLUDED_FROM_TESTING
-            )
-        ]
-
-        if untested:
-            pytest.fail(f"Documented but untested: {', '.join(sorted(untested))}")
+            specifier = Specifier(version_constraint)
+
+            # Check if any dependency in registry has min tested version matching the constraint
+            found_matching_dependency = False
+            for _, tested_range in tested_versions.items():
+                min_tested = tested_range.get("min")
+                if not min_tested:
+                    continue
+
+                constraint_major_minor = _get_major_minor(specifier.version)
+                tested_major_minor = _get_major_minor(min_tested)
+
+                if tested_major_minor == constraint_major_minor:
+                    found_matching_dependency = True
+                    break
+
+            if not found_matching_dependency:
+                constraint_major_minor = _get_major_minor(specifier.version)
+                expected_major_minor = f"{constraint_major_minor[0]}.{constraint_major_minor[1]}"
+
+                # we need to allowlist some special cases where we can't test the min version
+                if module_name in SPECIAL_CASES_ALLOWLIST:
+                    continue
+
+                errors.append(
+                    f"Integration '{integration_name}', Module '{module_name}': "
+                    f"No dependency found with min tested version matching supported constraint '{version_constraint}' "
+                    f"(expected major.minor: {expected_major_minor}). Minimum tested version: {min_tested}"
+                )
+
+    assert not errors, "\n".join(errors)
+
+
+def test_documented_vs_tested_versions(documented_versions: Dict[str, str], registry_data: list[dict]):
+    """Test that minimum documented versions align with minimum tested versions."""
+    registry_by_name = {entry["integration_name"]: entry for entry in registry_data}
+
+    misalignments = []
+
+    for integration_name, doc_version_spec in documented_versions.items():
+        registry_entry = registry_by_name.get(integration_name)
+        if not registry_entry or not registry_entry.get("is_tested", False):
+            continue
+
+        doc_version = _parse_version_spec(doc_version_spec)
+        registry_min_version = _get_registry_min_version(registry_entry)
+
+        if doc_version and registry_min_version and doc_version > registry_min_version:
```

Comment:
I'd think that the reverse condition is also important to know about: `doc_version < registry_min_version`

Reply:
Good catch! I added that condition, and it uncovered a few more inconsistencies, which have since been fixed!